File: kit/ListNode.go
package kit

// ListNode Definition for singly-linked list
type ListNode struct {
	Val  int
	Next *ListNode
}

// Ints2List Convert []int to List
func Ints2List(nums []int) *ListNode {
	emptyNode := &ListNode{}
	current := emptyNode
	for _, val := range nums {
		emptyNode.Next = &ListNode{Val: val}
		emptyNode = emptyNode.Next
	}
	return current.Next
}

File: kit/ListNode.js
// Definition for singly-linked list.
function ListNode(val) {
  this.val = val
  this.next = null
}

exports.ListNode = ListNode
exports.Ints2List = function(nums) {
  const emptyNode = new ListNode()
  nums.reduce((acc, cur) => {
    acc.next = new ListNode(cur)
    acc = acc.next
    return acc
  }, emptyNode)
  return emptyNode.next
}
